src/effects/RealtimeEffectStateUI.cpp

```diff
@@ -193,3 +198,36 @@ void RealtimeEffectStateUI::OnClose(wxCloseEvent & evt)
    if (next)
       next->ProcessEvent(evt);
 }
+
+void RealtimeEffectStateUI::ProjectWindowSubscriber::
+PushHandler(AudacityProject &project)
+{
+   auto& projectWindow = ProjectWindow::Get(project);
+   if (mState.mpProjectWindow == &projectWindow)
+      ;
+   else {
+      if (mState.mpProjectWindow)
+         mState.mpProjectWindow->RemoveEventHandler(this);
+      mState.mpProjectWindow = &projectWindow;
+      projectWindow.PushEventHandler(this);
+   }
+}
+
+BEGIN_EVENT_TABLE(RealtimeEffectStateUI::ProjectWindowSubscriber, wxEvtHandler)
+   EVT_CLOSE(RealtimeEffectStateUI::ProjectWindowSubscriber::InterceptCloseFrame)
+END_EVENT_TABLE()
+
+void RealtimeEffectStateUI::
+ProjectWindowSubscriber::InterceptCloseFrame(wxCloseEvent & evt)
+{
+   // Issue 4062, step 1:
```

Comment:
While I understand why `mpProjectWindow` is expected to be valid here, I would preffer to see a check or assertion before the first use